### PR TITLE
perf: skip DB query if requesting id only

### PR DESCRIPTION
**Pull request by Sekhmet**
## Summary

This PR will skip additional DB queries if only id is requested. New dependency had to be added to parse complex resolve info.

Closes https://github.com/snapshot-labs/checkpoint/issues/169

## Test plan

Change
```js
const item = await context.getLoader(returnType.name.toLowerCase()).load(id);
```

To
```js
console.log('querying', returnType.name.toLowerCase())
const item = await context.getLoader(returnType.name.toLowerCase()).load(id);
```

Run this request:
```gql
query {
  proposals {
    space {
      id
      strategies
    }
    author {
      id
    }
  }
}
```

You shouldn't see `querying User` in the console (only `querying Space`).

### CODE REVIEW

File: src/graphql/resolvers.ts [hunk 1]
```diff
@@ -69,12 +73,27 @@ export async function queryMulti(parent, args, context: ResolverContext, info) {
...
   const jsonFields = getJsonFields(returnType);
 
   const id = parent?.[info.fieldName] || args.id;
 
+  const parsed = parseResolveInfo(info);
+  if (parsed) {
+    // @ts-ignore
+    const simplified = simplifyParsedResolveInfoFragmentWithType(parsed, returnType);
+
+    if (Object.keys(simplified.fields).length === 1 && simplified.fields['id']) {
```
Comment (bonustrack) [inline src/graphql/resolvers.ts:92]: What if "id" is not the primary key?

Comment (bonustrack) [reply src/graphql/resolvers.ts:92]: Actually I dont think we support different primary key so this may not be relevant

Comment (Sekhmet) [reply src/graphql/resolvers.ts:92]: It has to be `id` as we have convention that when using nested entity it will be linked to that entity `id` field.

**Review by bonustrack — APPROVED** [review]
utACK

**Pull request merged**